app/scripts/modules/core/config/settings.ts

```diff
@@ -0,0 +1,174 @@
+interface IAzureProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IAWSProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamRole: string;
+    subnetType?: string;
+  };
+  defaultSecurityGroups: string[];
+  loadBalancers: {
+    inferInternalFlagFromSubnet: boolean;
+    certificateTypes?: string[];
+  };
+  useAmiBlockDeviceMappings: boolean;
+  classicLaunchLockout?: number;
+  metrics?: {
+    customNamespaces?: string[];
+  };
+}
+
+interface IGCEProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    zone: string;
+  };
+}
+
+interface ITitusProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamProfile: string;
+  };
+}
+
+interface IOpenStackProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IKubernetesProviderSettings {
+  defaults: {
+    account: string;
+    namespace: string;
+    proxy: string;
+  };
+}
+
+interface IAppengineProviderSettings {
+  defaults: {
+    account: string;
+    editLoadBalancerStageEnabled: boolean;
+  };
+}
```

Comment:
Not crazy about having provider-specific interfaces in the core module. Probably better to move these interfaces into the provider modules, then have a marker interface for `IProviderConfig` or something, with `IProviderSettings` below just being:
```
interface IProviderSettings {
  [key: string]: IProviderConfig
}
```

Does that make sense?